gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/tpgm/aggregation/AverageDuration.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.impl.operators.tpgm.aggregation;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.gradoop.common.model.impl.pojo.temporal.TemporalElement;
+import org.gradoop.common.model.impl.properties.PropertyValue;
+import org.gradoop.flink.model.api.tpgm.functions.TemporalAttribute;
+import org.gradoop.flink.model.api.tpgm.functions.aggregation.TemporalAggregateFunction;
+import org.gradoop.flink.model.impl.operators.aggregation.functions.BaseAggregateFunction;
+import org.gradoop.flink.model.impl.operators.aggregation.functions.average.Average;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Calculate the average duration of a time interval of temporal elements.
+ * Time intervals can be {@link TemporalAttribute}s.
+ * Time intervals with either the start or end time set to the respective default value will be
+ * ignored.
+ */
+public class AverageDuration extends BaseAggregateFunction
+  implements Average, TemporalAggregateFunction {
+
+  /**
+   * A property value containing the number {@code 1}, as a {@code long}.
+   */
+  private static final PropertyValue ONE = PropertyValue.create(1L);
```

Comment:
We can put this in _Average_ Interface and use it also in `AverageProperty`

Reply:
This would make this property public, it is currently only used to reduce `PropertyValue` instances.

Reply:
K